### ISSUE TITLE
Async/Sync bootstrap broken if validation key is missing (Knife 12.2 +)

### ISSUE THREAD

**Issue by marvelousNinja** [CenturyLinkCloud/clc-knife#65]
On QA machine with Chef DK (Knife version 12.5.1) we get the following error during sync bootstrap:
```
ERROR: You must pass a node name with -N when bootstrapping with user credentials
```
It seems that 12.5.1 and 12.6 might differ in terms of `:chef_node_name` option. We might need to generate default value by ourselves.

We might need to check async version too.

**Comment by marvelousNinja** [comment]
Investigation shown the following: QA machine was improperly configured (validation key was missing). However, we have a more serious problem here. In Chef 12.2 [validatorless bootstrap](https://www.chef.io/blog/2015/04/16/validatorless-bootstraps/) was introduced.

Basically, ```:validation_key``` parameter became optional. In order to connect to Chef server, bootstrap command generated keys on per-client basis. As a side effect - ```:chef_node_name``` became required (since node and client key for it should be generated **before** bootstrap script is executed on a target machine).

Implementing async validatorless bootstrap is surely out of the bug-fixing scope and should be implemented as a separate feature. For now, we should check for ```:validation_key``` before launching a server.

**Comment by marvelousNinja** [comment]
Fixed in #69 and #67.
